rawkit/raw.py

```diff
@@ -175,6 +176,52 @@ def save_thumb(self, filename=None):
         self.libraw.libraw_dcraw_thumb_writer(
             self.data, filename.encode('ascii'))
 
+    def get_4_col_raw(self):
+        """
+        Read the 4 colour raw data
+        The returned numpy array is (given colour RGGB)
+        ---------
+        R B R B ... 
+        G B G B ...
+        R B R B ....
+        . . . . .
+        . . . .  .
+        . . . .   .
+        --------
+
+
+        Returns:
+            numpy array: 4 colour data of the image in unit16. (width x height)
+            str : colour channel description (ie. RGGB, RGBG)
+        """
+        # Unpack the data, so that rawdata is populated
+        self.unpack()
+        rawdata = self.data.contents.rawdata
+
+        # Raise error if 4 color image isn't there, which will happen for some
+        # cameras
+        raise_if_error(rawdata.color4_image.contents == ctypes.c_voidp)
```

Comment:
`raise_if_error` was really made to be an internal (to the `libraw` package) helper method that deals with error codes. It shouldn't be used here. I'll open an issue to possibly update it to make this clearer, sorry about that.